contractcourt/chain_watcher.go

```diff
@@ -1160,14 +1150,31 @@ func (c *chainWatcher) dispatchContractBreach(spendEvent *chainntnfs.SpendDetail
 		closeSummary.LastChanSyncMsg = chanSync
 	}
 
-	if err := c.cfg.chanState.CloseChannel(
-		&closeSummary, channeldb.ChanStatusRemoteCloseInitiator,
-	); err != nil {
-		return err
+	// We create a function closure that will mark the channel as pending
+	// close in the database. We pass it to the contracBreach method such
+	// that it can ensure safe handoff of the breach before we close the
+	// channel.
+	markClosed := func() error {
```

Comment:
Why does this need to be passed in as a closure, rather than `cfg.contractBreach` taking `closeSummary` as a parameter and handling the close?

Reply:
I tried that approach at first, but figured it was a larger change, since we also would have to provide the `OpenChannel` to the breach arbiter. In addition we would need to handle the case there were it was added to the breach store, but not yet closed in the DB.

In other words, it was mostly to keep the diff smaller, but if it's preferable for understanding the code, maybe we should go with what you suggest?

Reply:
I was just wondering why, happy to leave as is.

> In addition we would need to handle the case there were it was added to the breach store, but not yet closed in the DB.

Particularly if we need to add an extra case if we don't pass the closure in, smaller diff sounds good :)

Reply:
Ok, keeping as is then :)